### PR TITLE
Add Shiden network

### PR DESCRIPTION
**Pull request by m00nbeans**
Adding the Shiden network (on Kusama).

### CODE REVIEW

File: src/networks.json [hunk 1]
```diff
@@ -498,5 +498,15 @@
...
     "rpc": ["https://rpc.moonriver.moonbeam.network"],
     "explorer": "https://blockscout.moonriver.moonbeam.network/"
+  },
+    "336": {
+    "key": "336",
+    "name": "Shiden (Kusama)",
+    "shortName": "Shiden",
+    "chainId": 336,
+    "network": "mainnet",
+    "multicall": "0x8AfCFB1757489B89c73fd2E0bd8fF0a5D985a8d9",
```
Comment (ChaituVR) [inline src/networks.json:508]: I am getting this error when i try to get balance of 0x8AfCFB1757489B89c73fd2E0bd8fF0a5D985a8d9 at block `1` maybe this is not a full archive node??
```
 (body="{\"jsonrpc\":\"2.0\",\"error\":{\"code\":-32603,\"message\":\"fetch runtime chain id failed: Application(Execution(Other(\\\"Exported method EthereumRuntimeRPCApi_account_basic is not found\\\")))\"},\"id\":44}\n", error={"code":-32603}, requestBody="{\"method\":\"eth_getBalance\",\"params\":[\"0x8afcfb1757489b89c73fd2e0bd8ff0a5d985a8d9\",\"0x1\"],\"id\":44,\"jsonrpc\":\"2.0\"}", requestMethod="POST", url="https://rpc.shiden.astar.network:8545", code=SERVER_ERROR, version=web/5.4.0)
 ```

**Review by ChaituVR — CHANGES_REQUESTED** [review]

**Comment by hskang9** [comment]
@ChaituVR The RPC provides full archive node link. This is not about a full archive node as I referred [here](https://github.com/snapshot-labs/snapshot.js/pull/388#discussion_r735101368). substrate is a upgradable blockchain where evm runtime does not start in block 0. full archive node link is the same. snapshot just needs to adjust listening all eth related rpc events starting from 339330.

**Review by bonustrack — APPROVED** [review]
@ChaituVR Can you merge this?

File: src/networks.json [hunk 2]
```diff
@@ -543,5 +543,15 @@
...
+    "rpc": ["https://rpc.shiden.astar.network:8545"],
+    "explorer": "https://shiden.subscan.io"
```
Comment (ChaituVR) [inline src/networks.json:555]: @m00nbeans @hskang9 I tested these again and i thiink `0x8AfCFB1757489B89c73fd2E0bd8fF0a5D985a8d9` is not a multicall strategy, 
I am getting the following error:
```
invalid type (argument="type", value="Call", code=INVALID_ARGUMENT, version=abi/5.4.1)
```
Make sure to pass a multicall contract here 
https://github.com/makerdao/multicall/blob/master/src/Multicall.sol
https://etherscan.io/address/0xeefba1e63905ef1d7acba5a8513c70307c1ce441#code

**Review by ChaituVR — CHANGES_REQUESTED** [review]

File: src/networks.json (hunk 2 above)
Comment (hskang9) [reply src/networks.json:555]: multicall in shiden address is 0xeb8B87169549d524cA4C60165E550f520A3A48e0. @ChaituVR Please update this

**Comment by bonustrack** [comment]
Can you add an image for the network here https://github.com/snapshot-labs/snapshot.js/tree/master/src/networks ?

**Comment by hskang9** [comment]
new PR with images are here https://github.com/snapshot-labs/snapshot.js/pull/435

**Pull request closed without merging**